File: trunk/AI/RAI/cpp-cgdk/model/PlayerContext.cpp
#include "PlayerContext.h"

using namespace model;
using namespace std;

PlayerContext::PlayerContext(const vector<Tank>& tanks, World world)
: tanks_(tanks), world_(world) { }

vector<Tank> PlayerContext::tanks() const {
	return tanks_;
}

World PlayerContext::world() const {
	return world_;
}

File: AI/RAI/java-cgdk/src/main/java/model/PlayerContext.java
package model;

import java.util.Arrays;

public final class PlayerContext {
    private final Tank[] tanks;
    private final World world;

    public PlayerContext(Tank[] tanks, World world) {
        this.tanks = Arrays.copyOf(tanks, tanks.length);
        this.world = world;
    }

    public Tank[] getTanks() {
        return Arrays.copyOf(tanks, tanks.length);
    }

    public World getWorld() {
        return world;
    }
}
